stx/mgr-restful-plugin

```diff
@@ -0,0 +1,321 @@
+DAEMON="/usr/bin/ceph-mgr"
+RUNDIR="/var/run/ceph/mgr"
+PIDFILE="${RUNDIR}/mgr-restful-plugin.pid"
+user_name=`hostname`
+MGR_ARGS="-f --cluster ceph --id ${user_name}"
+# The default 8003 port is already used by StarlingX "lsof -i :8003"
+# The orginal ceph-rest-api use 5001, so we keep using 5001 port here.
+PORT=5001
+REPLY_TIMEOUT_SECONDS=2
+
+restful_check()
+{
+# Steps:
+# 1) List all ceph mgr services
+#     {
+#         "restful": "https://nstcloud:5001/"
+#     }
+# 2) Extract restful plugin service
+#          "restful": "https://nstcloud:5001/"
+# 3) Remove the first colon
+#          "restful" "https://nstcloud:5001/"
+# 4) Remove all quote
+#           restful   https://nstcloud:5001/
+# 5) Split the result into an array
+    restful_rst=`ceph mgr services --connect-timeout $REPLY_TIMEOUT_SECONDS 2>&1 | \
+                 sed -n -e 's/.*\(\"restful\".*\)/\1/p' | \
+                 sed -n -e 's/://p' | \
+                 sed -n -e 's/\"//gp'`
+    IFS=' ' read -a restful_api <<< $restful_rst
+    if [ "pre_${restful_api[0]}" == "pre_restful" ]; then
+        return 0;
+    fi
+
+    return 1;
+}
+
+restful_service_disble_check()
+{
+   count=$1
+   i=0
+   while [ $i -lt $count ]
+   do
+       restful_check
+       rst=$?
+       if [ $rst -ne 0 ]; then
+           return 0
+       fi
+       sleep 1
+       i=$((i+1))
+   done
+   return $rst
+}
+
+mgr_active_check()
+{
+    ceph status --connect-timeout $REPLY_TIMEOUT_SECONDS 2>&1 | \
+    grep '\<mgr:' | grep -i '(active)'
+    rst=$?
+
+    if [ $rst -ne 0 ]; then
+        echo "no active mgr daemon"
+        return 1
+    else
+        echo "one mgr daemon is active"
+        return 0
+    fi
+}
+
+mgr_active_check_times()
+{
+    count=$1
+    i=0
+    rst=0
+    while [ $i -lt $count ]
+    do
+        mgr_active_check
+        rst=$?
+        if [ $rst -eq 0 ]; then
+            break
+        fi
+        i=$((i+1))
+        sleep 0.5
+    done
+
+    return $rst
+}
+
+mgr_check()
+{
+    if [ -f $PIDFILE ]; then
+        PIDDIR=/proc/$(cat $PIDFILE)
+        if [ -d ${PIDDIR} ]; then
+            comm_name=$(cat $PIDDIR/comm)
+            if [ $comm_name == $(basename $DAEMON) ]; then
+                echo "$(basename $DAEMON) is running"
+                return 0;
+            fi
+        fi
+    fi
+
+    echo "$(basename $DAEMON) is not running"
+    return 1;
+}
+
+start_restful_plugin()
+{
+    echo "Enabling ceph mgr restful plugin..."
+    ceph mgr module enable restful
+    rst=$?
+    if [ $rst -eq 0 ]; then
+        echo "ceph mgr restful plugin is enabled, need swith port to $PORT"
+    else
+        echo "ceph mgr restful plugin can't be enabled, abort"
+        return
+    fi
+
+    #ceph config-key set mgr/restful/${user_name}/server_port $PORT
+    ceph config-key set mgr/restful/server_port $PORT
+    ceph restful create-self-signed-cert
+    ceph restful create-key admin
+#    ceph mgr module enable restful
+    ceph restful restart
+    restful_check
```

Comment:
After "restful" is restarted either by "ceph restful create-self-signed-cert" or "ceph restful restart", we cannot assure "restful_check" will read out results within expected period of time.
So, you might have to reconsider the implementation of  "restful_check"!!